File: src/dataframe/rust/stats/distributions/students_t.test.R
#!/usr/bin/env Rscript

# Student's t distribution test script
# Usage: Rscript students_t.test.R <function> <x_or_p> <df> [lower_tail] [log_p]

source("test-helpers.R")

# Define Student's t distribution functions
students_t_density <- function(x, df, log = FALSE) {
  dt(x, df, log = log)
}

students_t_cumulative <- function(x, df, lower.tail = TRUE, log.p = FALSE) {
  pt(x, df, lower.tail = lower.tail, log.p = log.p)
}

students_t_quantile <- function(p, df, lower.tail = TRUE, log.p = FALSE) {
  qt(p, df, lower.tail = lower.tail, log.p = log.p)
}

students_t_random <- function(n, df) {
  rt(n, df)
}

# Run the test
run_distribution_test(students_t_density, students_t_cumulative, students_t_quantile, students_t_random)

File: src/dataframe/rust/stats/distributions/students_t.test.ts
#!/usr/bin/env -S deno run --allow-read --allow-run

import { callR, printResults, runComparison } from "./test-helpers.ts";

async function main() {
  console.log("Testing R vs Rust Student's t functions...\n");

  const testCases = [
    { func: "dt", distribution: "students_t", args: [1.5, 10, 0] },
    { func: "pt", distribution: "students_t", args: [1.5, 10, 1, 0] },
    { func: "qt", distribution: "students_t", args: [0.75, 10, 1, 0] },
  ];

  const results = await runComparison(testCases);
  printResults(results);

  // Test rt separately since it's not available in Rust interface
  console.log("\nR-only tests:");
  try {
    const rtResult = await callR("rt", "students_t", 1, 10);
    console.log(`rt(10) = ${rtResult}`);
  } catch (error) {
    console.log(`rt: Error - ${(error as Error).message}`);
  }
}

if (import.meta.main) {
  main().catch(console.error);
}
